dagshub/streaming/filesystem.py

```diff
@@ -106,33 +111,51 @@ def __init__(self,
 
         parsed_repo_url = urlparse(repo_url)
         content_api_path = f'/api/v1/repos{parsed_repo_url.path}/content/{branch}'
+        raw_api_path = f'/api/v1/repos{parsed_repo_url.path}/raw/{branch}'
 
         self.content_api_url = parsed_repo_url._replace(path=content_api_path).geturl()
-        self.raw_api_url = f'{repo_url}/raw/{branch}'
+        self.raw_api_url = parsed_repo_url._replace(path=raw_api_path).geturl()
         self.dvc_remote_url = f'{repo_url}.dvc/cache'
         self.dirtree = {}
 
-        del repo_url, branch, parsed_repo_url, content_api_path
+        del branch, parsed_repo_url, content_api_path
 
         # Determine if any authentication is needed
-        self.auth = (username, password) if username or password else None
-        del username, password
+        self.username = username if username else None
+        self.password = password if password else None
+        self.token = token if token else None
+
         response = self._api_listdir('')
         if response.ok:
-            # No authentication needed
             pass
         else:
-            # Check Git credential stores
-            proc = subprocess.run(['git', 'credential', 'fill'],
-                                input=f'url={repo_url}'.encode(),
-                                capture_output=True)
-            answer = {line[:line.index('=')]: line[line.index('=')+1:]
-                        for line in proc.stdout.decode().splitlines()}
-            if 'username' in answer and 'password' in answer:
-                self.auth = (answer['username'], answer['password'])
-            else:
-                # TODO: Check .dvc/config{,.local} for credentials
-                raise AuthenticationError('DagsHub credentials required, however none provided or discovered')
+            # TODO: Check .dvc/config{,.local} for credentials
+            raise AuthenticationError('DagsHub credentials required, however none provided or discovered')
+
+    @property
+    @lru_cache()
```

Comment:
If we're caching it here, then it needs a Time To Live or something similar because of expiration.